Replace architecture and vision documents with ADRs (#986)
itchysats/itchysats: docs/adr/007-event-sourcing-for-cfds.md

# ADR 007 - Use event sourcing for managing a CFD's state

```
Author: @thomaseizinger
Date: 29/12/2021
Status: Active
```

## Summary

Instead of recording the current state of the CFD and appending to that table, record the events that trigger the state transition and rebuild the current state on demand.

## Context

> Currently, we are always saving the _current_ state of a CFD in the database by appending to a `states` table.

Reviewer: ```suggestion
Previously, we were saving the _current_ state of a CFD in the database by appending to a `states` table.
```

Author: This was intentionally written in the context of when we made the decision: At that time of making the decision for event sourcing, we were saving the current state!

I think this is helpful if you'd read / catch up on historical decisions otherwise it is weird to come across decisions that seem to have been made with knowledge from the future.
